vignettes/DataPipeline.Rmd

```diff
@@ -22,132 +22,528 @@ knitr::opts_chunk$set(
-| dfFlagged | Flag         | The Flag value                      | Numeric  |
+Putting all of these steps together, the following script will create a sample report using both the analysis and reporting workflows for the AE KRI.
 
+```{r, eval = FALSE}
+lRaw <- list(
+    dfSUBJ = clindata::rawplus_dm,
+    dfAE = clindata::rawplus_ae,
+    dfPD = clindata::ctms_protdev,
+    dfLB = clindata::rawplus_lb,
+    dfSTUDCOMP = clindata::rawplus_studcomp,
+    dfSDRGCOMP = clindata::rawplus_sdrgcomp %>% dplyr::filter(.data$phase == 'Blinded Study Drug Completion'),
+    dfDATACHG = clindata::edc_data_points,
+    dfDATAENT = clindata::edc_data_pages,
+    dfQUERY = clindata::edc_queries,
+    dfENROLL = clindata::rawplus_enroll
+)
 
-## Appendix 2 - Workflow Specifications
+#make workflow lists
+wf_mapping <- MakeWorkflowList(strNames = "mapping")
+wf_kri <- MakeWorkflowList(strNames="kri0001")
+wf_reporting <- MakeWorkflowList(strNames = "reporting")
+
+# Generate Mapped Data
+lMapped <- RunWorkflows(lWorkflow = wf_mapping, lData = lRaw)
+
+# Generate Analysis Results Data
+lAnalysis <- RunWorkflows(lWorkflow = wf_kri, lData = lMapped)
+
+# Generate Reporting Data
+lReporting_Input <- list(
+    ctms_site = clindata::ctms_site,
+    ctms_study = clindata::ctms_study,
+    dfEnrolled =lMapped$dfEnrolled,
+    lWorkflows = wf_kri,
+    lAnalysis = list(kri0001 = lAnalysis),
+    dSnapshotDate = Sys.Date(),
+    strStudyID = "ABC-123"
+)
+
+lReporting <- RunWorkflows(lWorkflow = wf_reporting, lData = lReporting_Input)
+
+# Convenience Mappings
+dfGroups <- lReporting$dfGroups
+dfMetrics <- lReporting$dfMetrics
+dfSummary <- lReporting$dfSummary
+dfBounds <- lReporting$dfBounds
+
+# Create dfSite and dfStudy for charts
+dfSite <- dfGroups %>%
+  filter(GroupLevel == "Site") %>%
+  pivot_wider(names_from=Param, values_from=Value) %>%
+  rename(
+    SiteID = GroupID,
+    status = Status,
+    enrolled_participants = ParticipantCount
+  )
+
+dfStudy <-  dfGroups %>% 
+  filter(GroupLevel == "Study") %>% 
+  pivot_wider(names_from=Param, values_from=Value)
+
+#Create Charts for all metrics
+options(vsc.viewer = FALSE)
+metrics <- unique(dfMetrics$MetricID)
+charts <- metrics %>% map(~Visualize_Metric(
+  dfSummary = dfSummary,
+  dfBounds = dfBounds,
+  dfGroups = dfGroups,
+  dfMetrics = dfMetrics,
+  strMetricID = .x
+)
+) %>% setNames(metrics)
+
+##function not currently working!
+# Report_KRI(lCharts = charts,
+#            dfSummary = dfSummary, 
+#            dfSite = dfSite,
+#            dfStudy = dfStudy,
+#            dfMetrics = dfMetrics,
+#            strOutput = "~/test_report_kri.html")
+```
 
-Assessment workflow metadata objects are passed to the `lAssessments` parameter in `Study_Assess()` to define functions and parameters (including `mappings` and `specs`) across multiple studies.
+Additional examples are provided in the [Cookbook Vignette](https://gilead-biostats.github.io/gsm/articles/Cookbook.html).
```

Comment:
```suggestion
Additional examples are provided in the [Cookbook Vignette](Cookbook.html).
```
Or link in the same way as the previous one, if the magic to recode the link is intentional.